src/OrleansTestingHost/TestClusterOptions.cs

```diff
@@ -17,14 +19,62 @@ namespace Orleans.TestingHost
     [Serializable]
     public class TestClusterOptions
     {
+        /// <summary>
+        /// Extended options to be used as fallbacks in the case that explicit options are not provided by the user.
+        /// </summary>
+        public class FallbackOptions
+        {
+            /// <summary>
+            /// Is set to true, by default the cluster will output traces in the console
+            /// </summary>
+            public bool? TraceToConsole { get; set; }
+
+            /// <summary>
+            /// Default subfolder the the logs
+            /// </summary>
+            public string LogsFolder { get; set; }
+
+            /// <summary>
+            /// Default data connection string to use in tests
+            /// </summary>
+            public string DataConnectionString { get; set; }
+
+            /// <summary>
+            /// Default initial silo count
+            /// </summary>
+            public short InitialSilosCount { get; set; }
+
+            /// <summary>
+            /// Creates a default configuration builder with some defaults.
+            /// </summary>
+            /// <returns></returns>
+            public static ConfigurationBuilder DefaultConfigurationBuilder()
+            {
+                var builder = new ConfigurationBuilder();
+                builder.AddInMemoryCollection(new Dictionary<string, string>
+                {
+                    { nameof(DataConnectionString), "UseDevelopmentStorage=true" },
+                    { nameof(LogsFolder), "logs" },
+                    { nameof(TraceToConsole), "true" },
+                    { nameof(InitialSilosCount), "2" },
+                });
+                return builder;
+            }
+
+            /// <summary>
+            /// Configure defaults for building the configurations
+            /// </summary>
+            public static IConfiguration DefaultExtendedConfiguration { get; set; } = DefaultConfigurationBuilder().Build();
+        }
+
         private ClusterConfiguration _clusterConfiguration;
         private ClientConfiguration _clientConfiguration;
 
         /// <summary>
-        /// Create a new TestClusterOptions with the default initial silo count. See <see cref="DefaultInitialSilosCount"/>.
+        /// Create a new TestClusterOptions with the default initial silo count. See <see cref="ExtendedFallbackOptions"/>.
```

Comment:
Looks good to me. Except some minor comments. Testing code looks much cleaner now. I like it.